vscode/src/utilities/configuration.ts

```diff
@@ -0,0 +1,150 @@
+  return getConfigValue<number>("analysis.codeSnipLimit") || 10;
+}
+
+export function getConfigUseDefaultRulesets(): boolean {
+  return getConfigValue<boolean>("analysis.useDefaultRulesets") || true;
+}
+
+export function getConfigCustomRules(): string[] {
+  return getConfigValue<string[]>("analysis.customRules") || [];
+}
+
+export function getConfigLabelSelector(): string {
+  return getConfigValue<string>("analysis.labelSelector") || "discovery";
+}
+
+export function getConfigAnalyzeKnownLibraries(): boolean {
+  return getConfigValue<boolean>("analysis.analyzeKnownLibraries") || false;
+}
+
+export function getConfigAnalyzeDependencies(): boolean {
+  return getConfigValue<boolean>("analysis.analyzeDependencies") || true;
+}
+
+export function getConfigAnalyzeOnSave(): boolean {
+  return getConfigValue<boolean>("analysis.analyzeOnSave") || true;
+}
+
+export function getConfigDiffEditorType(): string {
+  return getConfigValue<string>("diffEditorType") || "diff";
+}
+
+export function getConfigKaiBackendURL(): string {
+  return getConfigValue<string>("kai.backendURL") || "0.0.0.0:8080";
+}
+
+export function getConfigKaiProviderName(): string {
+  return getConfigValue<string>("kai.providerName") || "ChatIBMGenAI";
+}
+
+export function getConfigKaiProviderModel(): string {
+  return getConfigValue<string>("kai.providerModel") || "meta-llama/llama-3-1-70b-instruct";
+}
+
+export function getConfigKaiProviderParameters(): object {
+  return getConfigValue<object>("kai.providerParameters") || {};
+}
+
+async function updateConfigValue<T>(
+  key: string,
+  value: T | undefined,
+  scope: vscode.ConfigurationTarget = vscode.ConfigurationTarget.Workspace,
+): Promise<void> {
+  await vscode.workspace.getConfiguration(KONVEYOR_CONFIG_KEY).update(key, value, scope);
+}
+
+export async function updateAnalyzerPath(value: string | undefined): Promise<void> {
+  await updateConfigValue("analyzerPath", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateKaiRpcServerPath(value: string | undefined): Promise<void> {
+  await updateConfigValue("kaiRpcServerPath", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateLogLevel(value: string): Promise<void> {
+  await updateConfigValue("logLevel", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateIncidentLimit(value: number): Promise<void> {
+  await updateConfigValue("analysis.incidentLimit", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateContextLines(value: number): Promise<void> {
+  await updateConfigValue("analysis.contextLines", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateCodeSnipLimit(value: number): Promise<void> {
+  await updateConfigValue("analysis.codeSnipLimit", value, vscode.ConfigurationTarget.Workspace);
+}
+
+export async function updateUseDefaultRulesets(value: boolean): Promise<void> {
```

Comment:
In the current naming scheme rule sets are joined using camel case ( `ruleSets`). Spell-checker reports that as an error which adds extra noise.
![image](https://github.com/user-attachments/assets/b8f178c2-bcb3-4a5f-9343-90479d213121)